src/utils/hooks/useGetResponsesAndElementState.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState, useEffect } from 'react'
+import { useGetElementsAndResponsesQuery, GetElementsAndResponsesQuery } from '../generated/graphql'
+import {
+  ResponsesByCode,
+  ResponsesFullByCode,
+  TemplateElementState,
+  EvaluatedElement,
+} from '../types'
+import evaluateExpression from '@openmsupply/expression-evaluator'
+
+const useGetResponsesAndElementState = (props: { serialNumber: string }) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const [responsesFullByCode, setResponsesFullByCode] = useState({})
+  const [elementsExpressions, setElementsExpressions] = useState([{}])
+  const [elementsState, setElementsState] = useState({})
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(true)
+  const { data, loading: apolloLoading, error: apolloError } = useGetElementsAndResponsesQuery({
+    variables: { serial: serialNumber },
+  })
+
+  useEffect(() => {
+    const error = checkForApplicationErrors(data)
+
+    if (error) {
+      setError(error)
+      setLoading(false)
+      return
+    }
+
+    if (apolloError) return
+
+    const applicationResponses = data?.applicationBySerial?.applicationResponses?.nodes
```

Comment:
```suggestion
    const applicationResponses = data?.applicationBySerial?.applicationResponses.nodes as ApplicationResponse[]
```

And add the import from `graphql`.

Reply:
OK